starlette/middleware/__init__.py

```diff
@@ -6,7 +6,7 @@ def __init__(self, cls: type, **options: typing.Any) -> None:
         self.cls = cls
         self.options = options
 
-    def __iter__(self) -> typing.Iterator:
+    def __iter__(self) -> typing.Iterator[typing.Any]:
```

Comment:
Is there really no better way to type this 😢 ?

Reply:
I've opened an issue on `mypy` regarding a related issue: https://github.com/python/mypy/issues/15408
The thing is `mypy` will use a "join" operation, finding the common supertype between `type` and `Dict[str, Any]`, which resolves to `object`. Maybe we could use `Iterator[Union[type, Dict[str, Any]]]` while ignoring the error or using `cast`

Reply:
Ok probably not worth worrying about too much. It seems like we’re just abusing `__iter__` for little benefit, but changing it would be a breaking change or at least larger refactor that’s not worth doing in this PR. Any is okay for now.